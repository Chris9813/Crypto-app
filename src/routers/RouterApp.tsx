import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from 'react-router-dom';
import { CoinsList } from '../components/Coins/CoinsList';
import { CoinsScreen } from '../components/Coins/CoinsScreen';
import { MarketList } from '../components/Market/MarketList';
import { Navbar } from '../components/ui/NavBar';


export const RouterApp = () => {
  return (
      <>
    <Router>
    <Navbar />
    <div>
        <Switch>
        <Switch>
                <Route exact path="/market/:marketId" component={ MarketList } />
                <Route exact path="/coins/:coinId" component={ CoinsScreen } />
                <Route path="/coins" component={ CoinsList } />
                <Route path="/market" component={MarketList} />
                <Redirect to="/coins" />
                </Switch>
        </Switch>
    </div>
</Router>
</>
  )
};
