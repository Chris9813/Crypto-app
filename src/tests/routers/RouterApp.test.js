import { shallow } from "enzyme";
import { RouterApp } from "../../routers/RouterApp";

describe("pruebas en el <RouterApp />", () => {
  test("debe mostrarse correctamente", () => {
    const wrapper = shallow(<RouterApp />);
    expect(wrapper).toMatchSnapshot();
  });
});
