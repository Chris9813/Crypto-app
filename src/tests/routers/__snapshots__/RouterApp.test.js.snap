// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en el <RouterApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <BrowserRouter>
    <Memo() />
    <div>
      <Switch>
        <Route
          component={[Function]}
          exact={true}
          path="/market/:marketId"
        />
        <Route
          component={[Function]}
          exact={true}
          path="/coins/:coinId"
        />
        <Route
          component={[Function]}
          path="/coins"
        />
        <Route
          component={[Function]}
          path="/exchanges"
        />
        <Redirect
          to="/coins"
        />
      </Switch>
    </div>
  </BrowserRouter>
</Fragment>
`;
